File: C++/condicionales.cpp
#include <iostream>
using namespace std;

int main() {
    int Figura;
    const double Pi = 3.1416;
    const int Const = 2;

    // Solicitar la figura al usuario
    cout << "Seleccione la figura para calcular su área:\n";
    cout << "1 para rombo\n";
    cout << "2 para rectángulo\n";
    cout << "3 para cuadrado\n";
    cout << "4 para trapecio\n";
    cout << "Ingrese su opción: ";
    cin >> Figura;

    double Area;

    switch (Figura) {
        case 1: { // Rombo
            double Dmayor, Dmenor;
            cout << "Ingrese el valor de la diagonal mayor: ";
            cin >> Dmayor;
            cout << "Ingrese el valor de la diagonal menor: ";
            cin >> Dmenor;
            Area = (Dmayor * Dmenor) / Const;
            cout << "El área del rombo es: " << Area << endl;
            break;
        }
        case 2: { // Rectángulo
            double Largo, Ancho;
            cout << "Ingrese el valor del largo del rectángulo: ";
            cin >> Largo;
            cout << "Ingrese el valor del ancho del rectángulo: ";
            cin >> Ancho;
            Area = Largo * Ancho;
            cout << "El área del rectángulo es: " << Area << endl;
            break;
        }
        case 3: { // Cuadrado
            double Lado;
            cout << "Ingrese el valor del lado del cuadrado: ";
            cin >> Lado;
            Area = Lado * Lado;
            cout << "El área del cuadrado es: " << Area << endl;
            break;
        }
        case 4: { // Trapecio
            double Bmayor, Bmenor, H;
            cout << "Ingrese el valor de la base mayor: ";
            cin >> Bmayor;
            cout << "Ingrese el valor de la base menor: ";
            cin >> Bmenor;
            cout << "Ingrese la altura del trapecio: ";
            cin >> H;
            Area = ((Bmayor + Bmenor) * H) / 2;
            cout << "El área del trapecio es: " << Area << endl;
            break;
        }
        default:
            cout << "Opción no válida." << endl;
    }

    return 0;
}

File: JAVA/condicionales.java
import java.util.Scanner;  // Importamos Scanner para leer la entrada del usuario

public class CalcularArea {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        final double PI = 3.1416; // Definir una constante
        final int CONST = 2; // Definir una constante entera

        // Pedir la figura
        System.out.println("Seleccione la figura a calcular su área: \n" +
                "1 para Rombo\n" +
                "2 para Rectángulo\n" +
                "3 para Cuadrado\n" +
                "4 para Trapecio\n");

        int figura = scanner.nextInt(); // Capturamos la opción

        switch (figura) {
            case 1: // Rombo
                System.out.print("Ingrese el valor de la diagonal mayor: ");
                int Dmayor = scanner.nextInt();
                System.out.print("Ingrese el valor de la diagonal menor: ");
                int Dmenor = scanner.nextInt();
                double areaRombo = (Dmayor * Dmenor) / (double) CONST;
                System.out.println("El área del rombo es: " + areaRombo);
                break;

            case 2: // Rectángulo
                System.out.print("Ingrese el valor del largo del rectángulo: ");
                int largo = scanner.nextInt();
                System.out.print("Ingrese el valor del ancho del rectángulo: ");
                int ancho = scanner.nextInt();
                int areaRectangulo = largo * ancho;
                System.out.println("El área del rectángulo es: " + areaRectangulo);
                break;

            case 3: // Cuadrado
                System.out.print("Ingrese el valor del lado del cuadrado: ");
                int lado = scanner.nextInt();
                int areaCuadrado = lado * lado;
                System.out.println("El área del cuadrado es: " + areaCuadrado);
                break;

            case 4: // Trapecio
                System.out.print("Ingrese el valor de la base mayor: ");
                int Bmayor = scanner.nextInt();
                System.out.print("Ingrese el valor de la base menor: ");
                int Bmenor = scanner.nextInt();
                System.out.print("Ingrese la altura del trapecio: ");
                int altura = scanner.nextInt();
                double areaTrapecio = ((Bmayor + Bmenor) * altura) / 2.0;
                System.out.println("El área del trapecio es: " + areaTrapecio);
                break;

            default:
                System.out.println("Opción inválida. Intente nuevamente.");
        }

        scanner.close(); // Cerrar el Scanner
    }
}
